### PR TITLE
Circulars Documentation: Markdown Primer

### LINKED ISSUE #1925: Add Markdown primer
It may be helpful to add a page to the documentation that explains what markdown is and the basic uses of it for circulars, something along the lines of [this](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) (but much more basic)

_Originally posted by @tylerbarna in https://github.com/nasa-gcn/gcn.nasa.gov/issues/1913#issuecomment-1944470132_

Add a primer under the Circulars documentation for Astro Flavored Markdown. It should include basic Markdown syntax and a quick reference to the cross-reference syntax that we support.

Here are some model Markdown primers:
- https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
- https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/quickstart-for-writing-on-github
- https://www.markdownguide.org/basic-syntax/
- https://daringfireball.net/projects/markdown/basics
- https://docs.gitlab.com/ee/user/markdown.html

We have a component that renders a quick reference for the cross-reference syntax:

https://github.com/nasa-gcn/gcn.nasa.gov/blob/0df1e7202a0c5a4daaf46e923536fe190dbe4d9a/app/routes/_gcn.circulars.edit.%24circularId/CircularEditForm.tsx#L73-L109

### PR DESCRIPTION
**Pull request by tylerbarna**
This PR adds a documentation page that explains markdown and how it can be used when writing new Circulars. Minor updates have also been made to existing documentation pages and components to reference the new page.

This PR resolves #1925

### CODE REVIEW

**Comment by lpsinger** [comment]
Please fix the build failure.

**Comment by tylerbarna** [comment]
> Please fix the build failure.

yeah sorry, I missed that the useFeature import was removed in another commit when I merged the upstream commits

**Comment by jracusin** [comment]
Can you please add some screenshots of the documentation?  It'll be easier to see how this all gets rendered that way.

**Comment by jracusin** [comment]
There also needs to be some explanation of how this interfaces to the automated linking and recognition of URLs that's already in place.

**Comment by tylerbarna** [comment]
> Can you please add some screenshots of the documentation? It'll be easier to see how this all gets rendered that way.

sure, here's the current version:
image
image
image
image
image
image

**Comment by lpsinger** [comment]
> This PR solves #1925

Remember to use [GitHub's keywords for automatically closing issues](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue).

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+<table class="usa-table">
+  <thead>
+    <tr>
+      <th>Example</th>
+      <th>Output</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>This is \*\*Bold\*\*</td>
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:32]: Please use inline code formatting for inline syntax examples.

```suggestion
    <td>`This is **Bold**`</td>
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+<table class="usa-table">
+  <thead>
+    <tr>
+      <th>Example</th>
+      <th>Output</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>
+      \-/\*/+ Item 1<br></br>\-/\*/+ Item 2<br></br>\-/\*/+ Item 3
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:150]: Please use block code formatting for block syntax examples.

```suggestion
      ```
      - Item 1
      * Item 2
      + Item 3
      ```
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      </table>
+    </td>
+  </tr>
+</table>
+
+### GCN Cross-References
+
+GCN also supports a cross-reference feature that allows for circulars to have links automatically generated for a number of commonly referenced sources, like DOIs, arXiv codes, and other circulars, when referenced in the body of a circular.
+
+{SyntaxReference()}
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:305]:
```suggestion
<SyntaxReference/>
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  </tr>
+  <tr>
+    <td>###### Header 6</td>
+    <td>
+      <h6>Header 6</h6>
+    </td>
+  </tr>
+</table>
+
+Header levels 1 and 2 can be created using an alternate method; the first and second levels of header is created by following the header text with a line of `=` or `-` symbols, respectively.
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:110]: Avoid passive voice.

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+      <pre>
+        <code>Code block line 1 Code block line 2</code>
+      </pre>
+    </td>
+  </tr>
+</table>
+
+### Tables
+
+Tables can be created in markdown by using the `|` symbol to separate columns and `-` symbols to separate the header from the body. The first row of the table is the header, and the second row is the separator. The rest of the rows are the body of the table.
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:263]: Markdown is a proper noun.

```suggestion
Tables can be created in Markdown by using the `|` symbol to separate columns and `-` symbols to separate the header from the body. The first row of the table is the header, and the second row is the separator. The rest of the rows are the body of the table.
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+
+export async function loader() {
+  if (!feature('CIRCULARS_MARKDOWN')) throw new Response(null, { status: 404 })
+  return null
+}
+
+# Astro-Flavored Markdown
+
+## Markdown Syntax
+
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:17]: Remove unnecessary heading. The preceding section has no content.

```suggestion
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 7]
```diff
@@ -0,0 +1,307 @@
+---
+handle:
+  breadcrumb: Markdown
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:3]:
```suggestion
  breadcrumb: Astro Flavored Markdown
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+
+import { feature } from '~/lib/env.server'
+import { SyntaxReference } from '~/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx'
+
+export async function loader() {
+  if (!feature('CIRCULARS_MARKDOWN')) throw new Response(null, { status: 404 })
+  return null
+}
+
+# Astro-Flavored Markdown
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:14]:
```suggestion
# Astro Flavored Markdown
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+
+## Markdown Syntax
+
+Markdown formatting can be accessed in the quick menu of the markdown bar, but it can also be added manually.
+
+### Basic Text Formatting
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:20]: Decrease levels of section headings.

```suggestion
## Basic Text Formatting
```

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+export async function loader() {
+  if (!feature('CIRCULARS_MARKDOWN')) throw new Response(null, { status: 404 })
+  return null
+}
+
+# Astro-Flavored Markdown
+
+## Markdown Syntax
+
+Markdown formatting can be accessed in the quick menu of the markdown bar, but it can also be added manually.
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:18]: This is not sufficiently descriptive of where to find this in the user interface.

**Review by lpsinger — CHANGES_REQUESTED** [review]
Would you please add some introductory text explaining what Markdown is, and linking to the first-party Markdown primer?

**Comment by tylerbarna** [comment]
> > This PR solves #1925
> 
> Remember to use [GitHub's keywords for automatically closing issues](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue).

oops, forgot 'solves' wasn't in the list. Thanks!

**Comment by tylerbarna** [comment]
It seems that the prettier checks are failing due to the block code syntax used in the block code examples

**Comment by lpsinger** [comment]
Please rebase.

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please fix build failures.

File: app/routes/_gcn.docs.circulars.markdown.mdx
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:1]: Rename this file to app/routes/docs.circulars.markdown.mdx.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+
+<table class="usa-table">
+  <thead>
+    <tr>
+      <th>Example</th>
+      <th>Output</th>
+    </tr>
+  </thead>
+  <tr>
+    <td># Header 1</td>
```
Comment (lpsinger) [inline app/routes/_gcn.docs.circulars.markdown.mdx:73]:
```suggestion
    <td>`# Header 1`</td>
```

...and same for the rest of the rows

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.docs.circulars.markdown.mdx [hunk 12]
```diff
@@ -0,0 +1,347 @@
...
+
+GCN supports the use of Markdown to format text and tables when writing Circulars. One can access a shortcut bar for quickly formatting text with Markdown by selecting the Markdown tab in the body section of the Circular submission web interface; the shortcut bar is then shown just above the text input area. This documentation describes the various formatting tools in the shortcut bar as well as the syntax that can be manually written to format text in Markdown.
+
+## Basic Text Formatting
+
+Markdown supports a number of basic text formatting options. Text is made bold or italic by enclosing it with `**` and `*` symbols, respectively. Attaching a link to specific text is accomplished by using the format `[Link](URL)`, and URLs and emails can be easily linked to without hyperlink text by enclosing them in angle brackets.
+
+<table class="usa-table">
```
[the remark this answers is not in the document]
Comment (tylerbarna) [reply app/routes/_gcn.docs.circulars.markdown.mdx]: I added some styling to the Table components to make them more in line with what's used for the SyntaxReference

**Review by lpsinger — APPROVED** [review]

**Comment by lpsinger** [comment]
Nice, thank you!

**Pull request merged**